### ISSUE TITLE
Cannot print test page for LaserJet 4000 or 2300

### ISSUE THREAD

**Issue by userhasaname** [OpenPrinting/ps-printer-app#4]
Thank you very much for your fix to the margin handling!  I updated the snap and I am now able to install the LaserJet 4000 ppd.  I do have an additional problem now, however.  I cannot print a test page to the printer.  If I print a test page, it starts, but does not complete.  The screen then says "No jobs in history."

I previously had a similar problem with the LaserJet 2300.  This time the test page starts printing but does not finish.  (Under jobs, it just says "Started at 16:49:01," but never progresses or actually prints.)  I then printed another test page which disappeared without printing (like the LaserJet 4000 does).

I am able to poll the options for each printer, so the application is communicating with them.  I also can print test pages using more generic ppds.  I have attached the log and state files.

[ps-printer-app.log(1).txt](https://github.com/OpenPrinting/ps-printer-app/files/6135672/ps-printer-app.log.1.txt)
[ps-printer-app.state(1).txt](https://github.com/OpenPrinting/ps-printer-app/files/6135673/ps-printer-app.state.1.txt)

**Comment by tillkamppeter** [comment]
If you click "Add Printer" in the web interface, do your printers appear under the auto-discovered devices ("Device:")? If so, select your auto-discovered printer, leave the "Driver Name:" field on "Auto-Detect" and create the printer. Does this printer print?

Also, before printing, make sure that the settings, "Device Settings", "Media", "Printing Defaults" correspond to the actual situation of your printer's installable accessories (like extra trays), the paper loaded in the trays, and finally the correct tray to take the paper for the test page from.

I assume that your printers are actually PostScript printers and not those which only get their PostScript capability only by an accessory (but I think their should be no problem as your printers answered the accessory and default option polls and they are also in (PostScript).

Could you send the test page directly to your printers using the following commands:
```
nc -w1 192.168.1.27 9100 < /snap/ps-printer-app/current/usr/share/ps-printer-app/testpage.ps
nc -w1 192.168.1.40 9100 < /snap/ps-printer-app/current/usr/share/ps-printer-app/testpage.ps
```
Does this work?

**Comment by tillkamppeter** [comment]
Your log file does not look like that you have printed a test page in that session. Please attach a log file which you obtain right after a failure of printing a test page.

**Comment by userhasaname** [comment]
Thanks!  My son is using the computer right now for Minecraft, but I will try these things tomorrow.  (Recreate log files, other steps, etc.)

Some quick answers though:  I did generate the log file (by copying it from the snap directory you gave me and then adding read permissions) after I tried to print the test page.  The test page never shows up in the print job history, so maybe that is part of the problem?

I was able to print the test page for both printers by creating generic postscript driver printers in the app, so I think sending the page with netcat will probably also work.

Both printers do have PostScript capabilities (and print with the more generic PostScript drivers.)

I think I have to install avahi-daemon or something to have the printers show up automatically through mdns.  On default Ubuntu 20.04 LTS, no printers show up as autodiscovered.  Also, when I tried "auto-detect" with the socket address before it wouldn't create a printer.

**Comment by userhasaname** [comment]
I printed the test page to the LaserJet 4000 using netcat and it worked.

I have a question on the logs.  If the application crashes and then restarts, are fresh logs created?  Do I need to attach the .O version of the logs in addition to the log and state?

**Comment by userhasaname** [comment]
It also works to print the test page to the LaserJet 2300 using netcat.

**Comment by tillkamppeter** [comment]
I do not investigate how PAPPL's logging logic works and what happens in case of a crash. But under the condition that systemd restarts the crashed Printer Application it can make the impression that the Printer Application did not actually crash. But you brought me to the idea now that your attempt to print a test page caused a crash and automatic restart with a new log being started (and hope the old one moved into .O), therefore you have a small log file (client count < 10).

Therefore I ask you first to try to print the test page again and then attach BOTH the current log and the ,O.

**Comment by userhasaname** [comment]
Thanks.  I will do that.  Is there also a systemd log I should attach?

**Comment by userhasaname** [comment]
Here are the logs that I copied after trying to print a test page.  Clicking the print test page button goes to a sort of incomplete version of the printer page, which I think probably indicated that it was printing.  (I was not quick enough to catch it.)  Then the printer page comes back up in the normal version showing no completed jobs.  (I have attached a screenshot of an earlier version of that, but as the time stamps indicated it was not immediately before I captured these logs.

[ps-printer-app.log.O.txt](https://github.com/OpenPrinting/ps-printer-app/files/6136805/ps-printer-app.log.O.txt)
[ps-printer-app.log.txt](https://github.com/OpenPrinting/ps-printer-app/files/6136806/ps-printer-app.log.txt)
[ps-printer-app.state.txt](https://github.com/OpenPrinting/ps-printer-app/files/6136807/ps-printer-app.state.txt)
![Screenshot_2021-03-14 LJ4000 - PostScript Printer Application(1)](https://user-images.githubusercontent.com/71998885/111071662-96d9d900-84ad-11eb-9eae-3cd1be86ecde.png)



.)

**Comment by tillkamppeter** [comment]
It was actually a crash, caused by the boolean option "ManualFeed" in the PPD file. My previous tests (~20 PPD files) all did not have such an option.

Should now be fixed with commit ee40f0f016. The Snap Store is rebuilding currently.

Thank you very much for your bug reports, the discovery of these two crashers was important.
